### ISSUE TITLE
`FactoryBot/CreateList` doesn't respect config

### ISSUE THREAD

**Issue by john-h-k** [rubocop/rubocop-factory_bot#64]
In the docs, `‘EnforcedStyle: create_list` (default)` is listed as the default style for `FactoryBot/CreateList`. However, this code:

```rb
my_assoc: [create(:bar, ...)]
```

causes this warning

```
C: [Correctable] FactoryBot/CreateList: Prefer 1.times.map.
        my_assoc: [create( ...
```

(Is it possibly seeing that `[create` is bad, but then suggesting `1.times.map` instead of `create_list`?)

**Comment by pirj** [comment]
Correct. Using create_list in certain cases had side effects and auto-correction is unsafe.
The cop thinks 1.times is less bad than a literal array.

**Comment by john-h-k** [comment]
I don't think that is true. The autocorrect for `CreateList` is marked as safe and it does it whenever there is more than 1 element in the list

```rb
[create(:foo), create(:foo)]
```

is autocorrected to

```rb
create_list(:foo, 2)
``

There is a [comment in the cop itself](https://github.com/rubocop/rubocop-factory_bot/blob/afc310efc27661d673349aa560b3559dae1ab7b5/lib/rubocop/cop/factory_bot/create_list.rb#L157) which I think gives a clue to what is happening.

```
# For ease of modification, it is replaced with the `n_times` style,
# but if it is not appropriate for the configured style,
# it will be replaced in the subsequent autocorrection.
```

So normally, I am inferring, it replaces `[create(:foo), create(:foo)]` with `2.times.map { create(:foo) }`, and then autocorrects that to `create_list(:foo, 2)`.

My guess for why it doesn't work when n=1 is that `1.times.map` triggers an unrelated rubocop, [`Lint/UselessTime`](https://docs.rubocop.org/rubocop/cops_lint.html#lintuselesstimes), which is somehow blocking the second part of autocorrection happening.

I would be more than happy to fix the cop so it properly handles the n=1 case


> The cop thinks 1.times is less bad than a literal array.

Apart from the above bug, why is this? `1.times.map` is more characters, slower, less intuitive, and more bugprone (as well as triggering a rubocop)

**Comment by pirj** [comment]
From this example which is similar to yours https://github.com/rubocop/rubocop-factory_bot/blob/afc310efc27661d673349aa560b3559dae1ab7b5/spec/rubocop/cop/factory_bot/create_list_spec.rb#L242, there should be no offence.
Can you debug why there is?

**Comment by ahorner** [comment]
I was running into the same issue, so I dug into this a bit. It looks like this was actually addressed by https://github.com/rubocop/rubocop-factory_bot/commit/e7a68696c74fef53804efabeb229b9f54ef3f691, which is where the test that @pirj linked was updated. 

The issue is that this commit isn't included in any published versions of the gem, so you'll need to point to the repo for the gem source until a new version is cut.

**Comment by pirj** [comment]
Nice dig, thanks!

@ydah would you like to re-apply that PR? It seems it has been missed somehow.

**Comment by ydah** [comment]
@pirj Is this PR? Perhaps it has been merged into the master branch, so releasing it might solve the issue.
- https://github.com/rubocop/rubocop-factory_bot/pull/48

**Comment by pirj** [comment]
@ahorner @john-h-k can you please check if the problem is solved if you use the ‘master’ branch?

**Comment by ahorner** [comment]
@ydah: Looks good! I'm able to use the official release again, thank you.

**Comment by ydah** [comment]
@ahorner Thank you!
